sycl/include/sycl/detail/cg_types.hpp

```diff
@@ -69,6 +69,101 @@ enum class CGType : unsigned int {
   AsyncFree = 29,
 };
 
+// The structure represents NDRange - global, local sizes, global offset and
+// number of dimensions.
+
+// TODO: A lot of tests rely on particular values to be set for dimensions that
+// are not used. To clarify, for example, if a 2D kernel is invoked, in
+// NDRDescT, the value of index 2 in GlobalSize must be set to either 1 or 0
```

Comment:
I really dislike this aspect of the NDRDesc.  I has THREE different applications:
 - sometimes we template it to Dims
 - sometimes Dim is 3, but 0 are used in the higher "unused" dimensions
 - othertimes Dim is 3 and 1 is used in the higher "unused" dimensions.
 
IIRC, when we are hoisting it to the device it is not templated and we use that third convention.  That way the sizes are correct, as multiplying by 1 doesn't change anything.  

We should probably remove that second convention, the one that uses 0.  That just leads to bugs. And, maybe we should reevaluate the usage of this end to end and drop the template over Dims. In one form or other, the NDRDesc is used by the FE, by SYCL and by the device.